pkg/util/kernel/find_headers.go

```diff
@@ -110,19 +111,22 @@ func GetKernelHeaders(headerDirs []string, headerDownloadDir, aptConfigDir, yumR
 		}
 	}
 
-	d := headerDownloader{aptConfigDir, yumReposDir, zypperReposDir}
-	if err = d.downloadHeaders(headerDownloadDir); err != nil {
-		if errors.Is(err, errReposDirInaccessible) {
-			return nil, reposDirAccessFailure, fmt.Errorf("unable to download kernel headers: %w", err)
+	if downloadEnabled {
+		d := headerDownloader{aptConfigDir, yumReposDir, zypperReposDir}
+		if err = d.downloadHeaders(headerDownloadDir); err != nil {
+			if errors.Is(err, errReposDirInaccessible) {
+				return nil, reposDirAccessFailure, fmt.Errorf("unable to download kernel headers: %w", err)
+			}
+			return nil, downloadFailure, fmt.Errorf("unable to download kernel headers: %w", err)
 		}
-		return nil, downloadFailure, fmt.Errorf("unable to download kernel headers: %w", err)
-	}
 
-	log.Infof("successfully downloaded kernel headers to %s", headerDownloadDir)
-	if err = validateHeaderDirs(hv, dirs); err == nil {
-		return dirs, downloadSuccess, nil
+		log.Infof("successfully downloaded kernel headers to %s", headerDownloadDir)
+		if err = validateHeaderDirs(hv, dirs); err == nil {
+			return dirs, downloadSuccess, nil
+		}
+		return nil, validationFailure, fmt.Errorf("downloaded headers are not valid: %w", err)
 	}
-	return nil, validationFailure, fmt.Errorf("downloaded headers are not valid: %w", err)
+	return nil, headersNotFound, fmt.Errorf("kernel header downloading is disabled")
```

Comment:
This will log an error, right? If so, it might be worth trying to log it as a warning to not confuse customers

Reply:
Maybe just a rewording of the error message?

Reply:
That works I think 👍